config/config-network.yaml

```diff
@@ -173,12 +173,25 @@ data:
     # Knative doesn't know about that otherwise.
     default-external-scheme: "http"
 
-    # internal-encryption indicates whether internal traffic is encrypted or not.
-    # If this is "true", the following traffic are encrypted:
-    #  - ingress to activator
-    #  - ingress to queue-proxy
-    #  - activator to queue-proxy
+    # internal-encryption applies to the knative control-plane only
+    # internal-encryption use for data-plane is deprecated and is replaced by internal-trust
+    # internal-encryption indicates whether internal control-plane traffic is encrypted or not.
     #
     # NOTE: This flag is in an alpha state and is mostly here to enable internal testing
     #       for now. Use with caution.
     internal-encryption: "false"
+
+    # internal-trust indicates the level of trust established in the knative data-plane.
+    # internal-trust = "none" (the default) - uses no encryption for internal traffic
+    # Using any other value ensures that the following traffic are encrypted:
+    #  - ingress to activator
+    #  - ingress to queue-proxy
+    #  - activator to queue-proxy
+    #
+    # internal-trust = "minimal" ensures requests arriving the Kingress will be forwarded to Kserv with a certificate created by the Knative CA
+    # internal-trust = "namespace" ensures requests arriving the Kingress will be forwarded to Kserv with a namespace specific certificate created by the Knative CA
+    # internal-trust = "mutual" same as "namespace" and also ensures that Kserv will only receive reqeusts via the Kingress
+    # internal-trust = "identity" same as "mutual" and also adds sender identity verified by the Kingress and forwarded to Kserv
+    #
+    # NOTE: This flag is in an alpha state and is mostly here to enable internal testing for now. Use with caution.
+    internal-trust: "none"
```

Comment:
I'm not sure that "trust" is the right word here -- maybe you mean "internal-auth" or "internal-verification"?  We're _establishing_ trust, but the variable name could also be seen as _declaring_ trust without an enforcement mechanism.

Reply:
> I'm not sure that "trust" is the right word here -- maybe you mean "internal-auth" or "internal-verification"? We're _establishing_ trust, but the variable name could also be seen as _declaring_ trust without an enforcement mechanism.

It is ok to change trust, but it seems better than auth even verification. 
since it is suggested to control if we use any description and if we do use encryption, then, verifications do we do.